### PR TITLE
Reexport

### PR DESCRIPTION
**Pull request by Licenser**
This is an example how we could handle re-exporting so we don't need to copy from `std::arch` (avoiding possible type conflicts) and users don't need to import both `std::arch` and `simd-lite`. What do you think?

### CODE REVIEW

**Comment by sunnygleason** [comment]
Nice, I like where this is going!!!

I wonder if we could make something work along these lines?

* simd-lite declares the core simd types which are essentially copies++ of stdarch types
* simd-lite provides nicer "sugar" methods like `::new()`, array indexing, comparison operations, & more...
* simd-lite provides From/Into (or equivalent) traits/functions for seamless translation to stdarch types
* that way, the simd-lite functions have a uniform interface (and sugar), but can translate from/into stdarch types?

That might be a really nice interim step as we figure out how
to target stable rust. And then once that's possible, developers
have no additional work to do because the API is just about
identical.

Anyway, I don't even know if this is possible, but it seemed like
a clear way to have a value-add for the library beyond 2-3 dozen
intrinsics that developers may or may not need...

**Comment by Licenser** [comment]
The problem with re-declaring the types remains, it will make it incompatible with stdlib so anyone using simd-lite and std will have a whole lot of pain to deal with with `mem::transmute` or some other conversion all over the place :/

**Comment by sunnygleason** [comment]
@Licenser the type is incompatible 1:1, but is there anything we can do with From/Into or other traits to make them more easily interchangeable?

**Comment by Licenser** [comment]
Ja rust (sadly/fortunately?) doesn't work that way, if even if type `crate_a::type` and `crate_b::type` are bit identical they're they are treated as entirely different types and without some kind of conversion one can't be transfered into another.

That might not be so bad for NEON given that `std` implements virtually no neon at the moment but for the AVX*'s it'd get quite annoying unless we literally want to clone `std::arch`.

Imagine the following code:

```rust
use std::arch::x86_64::*;
use simd_lite::x86_64::*;

unsafe fn something(from: __mm128)  -> __mm512 {
    let to = _mm512_castpd128_pd512(from);
   _mm512_abs_epi16(to)
}
```

Now a user would need to think about:
1) If is `_mm512_castpd128_pd512` implemented by us, then `from` needs to be `simd-lite::__mm128` otherwise it needs to be `std::...::__mm128`
2) only `_mm512_castpd128_pd512` or only `_mm512_abs_epi16` is implemented by us then the `(to)` needs to be `(to.into())` (or equivalent cast); if both er from simd-lite or std then this is not the case.
3) if the result of the function is used with std code then `-> __mm512` should be `std::..::__mm512` otherwise `simd-llite::__mm512`
4) if `_mm512_abs_epi16` is implemented by us and the result from 3) is `std::..::__mm512` then it needs to be `_mm512_abs_epi16().into()` the same if `_mm512_abs_epi16` is implemented in std and 3) is `simd-llite::__mm512` but if both are from the same crate it doesn't.

That of cause could be worked around if we copy every single function and type in `std::arch` but then the same problem re-alears if you have two crates one using `std` the other using simd-lite` and the problem described above resurfaces just the same way.

Does my reservation of cloning datatypes make sense in that context?

I am pretty srue we can implement a lot of the niceties with a trait that we implement for the `std::` types - the same way `::new()` for the types works right now.

**Comment by sunnygleason** [comment]
@Licenser I see what you're saying, and can definitely relate to that
viewpoint. It is definitely desirable to be respectful/cooperative with
stdarch APIs, I think I've just lost faith that stdarch is going to be useful
enough quickly enough based on its current progress & contribution
"manners" and philosophy.

I think that as this library gets expanded, especially if we keep the "drop-in"
philosophy, people will have less reason to use it alongside stdarch. If we
can get the automated LLVM parsing working, I don't think we're far off
from being able to have a large portion of LLVM linked soon for x86/x64
and arm/aarch64.

But even given the functionality, I can see how adoption would be a
long way from that! So I definitely don't want to hold up this PR, I think
it's good! Let me know if you need any concrete help/review. This seems
nicely done, I'd only look for more tests (unless I missed them?) to make
sure we don't break things with future code contributions.

**Comment by Licenser** [comment]
Ja the conversation just happened to be in here :P. I absolutely understand the loss on confidence in stdarch and share it. I usually try to aim for the principle of least suprise, and copying types / creating incompatibilities kind of violates that to me. What I wonder is what features couldn't be provided by a trait that you would like to see?

I'm happy to just template on how far we can get w/o copying on this branch and we compare at the end, I think with the re-export as it is we get nearly everything  (I can't think of what would be missing) of what we want without the pain. (but vacation is over so time is a bit more sparse :P)

**Comment by Licenser** [comment]
Okay I marked this as ready for merge, while there are technically no separate tests for ::new() and some of the vld1 instructions they are implicitly in every test. I think that's good enough ™️ for the time being so we can crate this? I'll add a ticket so we don't forget to back-fill them later.

File: build.rs [hunk 1]
```diff
@@ -204,7 +204,7 @@ fn values(t: &str, vs: &[String]) -> String {
         format!(": {} = {}", t, vs[0])
     } else {
         format!(
-            ": {} = transmute([{}])",
+            " = {}::new([{}])",
```
Comment (sunnygleason) [inline build.rs:207]: :+1:!

File: src/aarch64/neon/tests.rs [hunk 2]
```diff
@@ -14,480 +13,15 @@ macro_rules! transmute {
 
 #[test] //FIXME: #[simd_test(enable = "neon")]
 fn test_vpaddq_u8() {
-    let a = i8x16::new(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15);
-    let b = i8x16::new(
+    let a = int8x16_t::new([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]);
+    let b = int8x16_t::new([
         17, 18, 19, 20, 20, 21, 22, 23, 24, 25, 26, 27, 29, 29, 30, 31,
-    );
```
Comment (sunnygleason) [inline src/aarch64/neon/tests.rs:20]: nice! I was thinking at some point we might want to add in similar tests as "smoke" tests. Will add an issue for that...

File: src/simd.rs [hunk 3]
```diff
@@ -1,204 +0,0 @@
-//! Internal `#[repr(simd)]` types
```
Comment (sunnygleason) [inline src/simd.rs:1]: 👋 bye!

**Review by sunnygleason — APPROVED** [review]
@Licenser this looks great, thank you!

**Pull request merged**